cc/build_defs.bzl

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright 2022 The Project Oak Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Provides a rule that outputs a monolithic static library."""
+
+load(
+    "@bazel_tools//tools/cpp:toolchain_utils.bzl",
+    "find_cpp_toolchain",
+    "use_cpp_toolchain",
+)
+
+# Bazel rule for collecting the header files that a target depends on.
+def _cc_static_library_impl(ctx):
```

Comment:
How does this rule handle [copts](https://bazel.build/reference/be/c-cpp#cc_library.copts) and [defines](https://bazel.build/reference/be/c-cpp#cc_library.defines) ?

Reply:
It doesn't need to handle copts and defines as it's working as a wrapper :)

Use current codebase as example:
```
cc_library(
    name = "tflite-micro",
    srcs = ["tflite_micro.cc"],
    hdrs = ["tflite_micro.h"],
    linkstatic = 1,
    deps = [
        "@com_github_tensorflow_tflite_micro//tensorflow/lite/micro:micro_framework",
    ],
)

cc_static_library(
    name = "tflite-micro-fully-static",
    deps = [":tflite-micro"],
)
```
The `tflite-micro-fully-static` target searches all required libraries for `tflite-micro`, then uses `ar` to archive all of them into one archive a.k.a. fully static library.
In this case, all *.a generated from tflite-micro as well as @com_github_tensorflow_tflite_micro//tensorflow/lite/micro:micro_framework are `ar` into a fully static `libtflite-micro-fully-static.a`.